src/Lecture1.hs

```diff
@@ -119,4 +147,16 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list = 
+    let calculateMin x acc l = if null l then acc else calculateMin x (if (head l) < x then (acc + 1) else acc) (tail l)
```

Comment:
I see GHC warnings about defaulting values. This happens because Haskell infers the most polymorphic type and numbers are polymorphic hence the result of `calculateMin` and `calculateMax` functions is also some polymorphic value.

You can fix this by specifying the type of functions in explicitly. You can do this in `let` as well:


```suggestion
    let calculateMin :: Int -> Int -> [Int] -> Int
        calculateMin x acc l = if null l then acc else calculateMin x (if (head l) < x then (acc + 1) else acc) (tail l)
```